packages/data-service/README.md

```diff
@@ -0,0 +1,147 @@
+This package is to be extracted to dcos-ui/data-service.
+
+# API v1
+
+## Mediator Pattern
+
+The [mediator pattern](https://en.wikipedia.org/wiki/Mediator_pattern) is used here
+to decouple data handling from the view components by encapsulating the way they
+communicate. 
+
+With this pattern, a component will receive down all the data `(data down)` it needs via props,
+and send `events up` so the mediator can perform mutations (represented here as a simple callback). 
+The key concept here is that both the data provider and the view component have no idea how each other work.
+Only the caller of `renderComponents` understands how it works.  
+
+```js
+  /*
+   * Receives a stream of React.Component and outputs a single components that 
+   * renders the most recevent event from the Observable.
+  */
+  componentFromStream(component$: Observable<React.Component>): React.Component;
+```
+
+Here is an example of how it could be used to render a page for `space shuttles`
+that shows which shuttles are on space, and which are ready to be launched.
+
+Note that the way we achieve the Mediator pattern is not in an OO fashion, but
+by abstracting all the communication between the data sources and the view components
+by using a stream.
+
+## Query Example
+
+To exemplify how you would use the Mediator pattern to get data from a graphql
+source and render a React component we extracted this is a working example from 
+[DC/OS UI](https://github.com/dcos/dcos-ui/).
+
+```js
+import React from "react";
+import { Observable } from "rxjs";
+
+// Recompose helps handle data down and event up as streams
+import {
+  createEventHandler,
+  componentFromStream
+} from "#PACKAGES/data-service/recomposeRx";
+
+// Our in-house solution to resolve graphql queries using streams
+import { graphqlObservable } from "#PACKAGES/data-service/graphqlObservable";
+
+// tools to make easier to create schemas and queries
+import { makeExecutableSchema } from "graphql-tools";
+import gql from "graphql-tag";
+
+// Streams we get our data from
+import { fetchRepositories } from "#PLUGINS/catalog/src/js/repositories/repositoriesStream";
+
+// The graphql schema and resolvers for those streams;
+import { typeDefs, resolvers } from "#SRC/js/model/repositoryModel";
+
+// UI components
+import RepositoriesTabUI from "#SRC/js/components/RepositoriesTabUI";
+import RepositoryList from "#SRC/js/structs/RepositoryList";
+
+// Using the data layer
+
+// 1. We first make an schema out of the resolvers and typeDefinitions
+const schema = makeExecutableSchema({
+  typeDefs,
+  resolvers
+});
+
+// 2. We declare which data our component needs in graphql (data down)
```

Comment:
It is assumed you know graphql and rxjs. Otherwise I would be written a bunch of stuff the internet does much better. I added the schema and the resolvers as asked on the previous comment, that should be enough to understand given those assumptions.